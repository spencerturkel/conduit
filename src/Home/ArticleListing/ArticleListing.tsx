import * as React from 'react';
import {ComponentType, StatelessComponent} from 'react';

export interface ArticlePreview {
    author: {
        image: string;
        username: string;
    };
    createdAt: Date;
    description: string;
    favoritesCount: number;
    slug: string;
    title: string;
}

export interface ArticleListingProps {
    previews: ArticlePreview[];
}

const ArticleListing = (FeedPicker: ComponentType): StatelessComponent<ArticleListingProps> => ({previews}) => (
    <>
        <FeedPicker />

        {previews.map(preview => (
            <div key={preview.slug} className="article-preview">
                <div className="article-meta">
                    <a href={'/profile/' + preview.author.username}>
                        <img src={preview.author.image} />
                    </a>
                    <div className="info">
                        <a href={'/profile/' + preview.author.username} className="author">
                            {preview.author.username}
                        </a>
                        <span className="date">{preview.createdAt.toDateString()}</span>
                    </div>
                    <button className="btn btn-outline-primary btn-sm pull-xs-right">
                        <i className="ion-heart" /> {preview.favoritesCount}
                    </button>
                </div>
                <a href={'/article/' + preview.slug} className="preview-link">
                    <h1>{preview.title}</h1>
                    <p>{preview.description}</p>
                    <span>Read more...</span>
                </a>
            </div>
        ))}
    </>
);

export default ArticleListing;
