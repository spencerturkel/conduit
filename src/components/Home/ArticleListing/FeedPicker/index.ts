export { default } from "./FeedPicker";
export * from "./FeedPicker";
