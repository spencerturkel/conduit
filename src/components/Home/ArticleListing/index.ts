export {default} from './ArticleListing';
export * from './ArticleListing';
