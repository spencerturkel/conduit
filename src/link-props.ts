import * as React from 'react';

export interface LinkProps {
    children?: React.ReactNode;
    className: string;
    to: string;
}
