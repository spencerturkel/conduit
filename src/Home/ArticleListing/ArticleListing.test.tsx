import {mount, ReactWrapper} from 'enzyme';
import * as React from 'react';
import {StatelessComponent} from 'react';

import {LinkProps} from '../../link-props';
import ArticleListing, {ArticlePreview} from './ArticleListing';

const TestFeedPicker = () => <div>Feed Picker</div>;
const TestLink: StatelessComponent<LinkProps> = ({to: href, ...rest}) => <a {...{href}} {...rest} />;
const TestArticleListing = ArticleListing(TestFeedPicker, TestLink);

const testProps: ArticlePreview[] = [
    {
        author: {
            image: 'https://i.stack.imgur.com/xHWG8.jpg',
            username: 'jake',
        },
        createdAt: new Date('2017-10-12'),
        description: 'This is the description for the post.',
        favoritesCount: 27,
        slug: 'test-slug',
        title: 'How to build webapps that scale',
    },
];

let sut: ReactWrapper;

beforeEach(() => {
    sut = mount(<TestArticleListing previews={testProps} />);
});

it('should match snapshot', () => {
    expect(sut).toMatchSnapshot();
});
