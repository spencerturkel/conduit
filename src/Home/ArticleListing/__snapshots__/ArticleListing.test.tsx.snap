// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <Unknown
    previews={
        Array [
            Object {
              "author": Object {
                "image": "https://i.stack.imgur.com/xHWG8.jpg",
                "username": "jake",
              },
              "createdAt": 2017-10-12T00:00:00.000Z,
              "description": "This is the description for the post.",
              "favoritesCount": 27,
              "slug": "test-slug",
              "title": "How to build webapps that scale",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "previews": Array [
        Object {
          "author": Object {
            "image": "https://i.stack.imgur.com/xHWG8.jpg",
            "username": "jake",
          },
          "createdAt": 2017-10-12T00:00:00.000Z,
          "description": "This is the description for the post.",
          "favoritesCount": 27,
          "slug": "test-slug",
          "title": "How to build webapps that scale",
        },
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": Object {
          "instance": <div>
            Feed Picker
</div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Feed Picker",
          },
          "ref": null,
          "rendered": Array [
            "Feed Picker",
          ],
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": <div
          class="article-preview"
>
          <div
                    class="article-meta"
          >
                    <a
                              href="/profile/jake"
                    >
                              <img
                                        src="https://i.stack.imgur.com/xHWG8.jpg"
                              />
                    </a>
                    <div
                              class="info"
                    >
                              <a
                                        href="/profile/jake"
                                        class="author"
                              >
                                        jake
                              </a>
                              <span
                                        class="date"
                              >
                                        Wed Oct 11 2017
                              </span>
                    </div>
                    <button
                              class="btn btn-outline-primary btn-sm pull-xs-right"
                    >
                              <i
                                        class="ion-heart"
                              />
                              27
                    </button>
          </div>
          <a
                    href="/article/test-slug"
                    class="preview-link"
          >
                    <h1>
                              How to build webapps that scale
                    </h1>
                    <p>
                              This is the description for the post.
                    </p>
                    <span>
                              Read more...
                    </span>
          </a>
</div>,
        "key": "test-slug",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="article-meta"
>
              <a
                            href="/profile/jake"
              >
                            <img
                                          src="https://i.stack.imgur.com/xHWG8.jpg"
                            />
              </a>
              <div
                            className="info"
              >
                            <a
                                          className="author"
                                          href="/profile/jake"
                            >
                                          jake
                            </a>
                            <span
                                          className="date"
                            >
                                          Wed Oct 11 2017
                            </span>
              </div>
              <button
                            className="btn btn-outline-primary btn-sm pull-xs-right"
              >
                            <i
                                          className="ion-heart"
                            />
                             
                            27
              </button>
</div>,
            <a
              className="preview-link"
              href="/article/test-slug"
>
              <h1>
                            How to build webapps that scale
              </h1>
              <p>
                            This is the description for the post.
              </p>
              <span>
                            Read more...
              </span>
</a>,
          ],
          "className": "article-preview",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="article-meta"
>
              <a
                            href="/profile/jake"
              >
                            <img
                                          src="https://i.stack.imgur.com/xHWG8.jpg"
                            />
              </a>
              <div
                            class="info"
              >
                            <a
                                          href="/profile/jake"
                                          class="author"
                            >
                                          jake
                            </a>
                            <span
                                          class="date"
                            >
                                          Wed Oct 11 2017
                            </span>
              </div>
              <button
                            class="btn btn-outline-primary btn-sm pull-xs-right"
              >
                            <i
                                          class="ion-heart"
                            />
                            27
              </button>
</div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  href="/profile/jake"
>
                  <img
                                    src="https://i.stack.imgur.com/xHWG8.jpg"
                  />
</a>,
                <div
                  className="info"
>
                  <a
                                    className="author"
                                    href="/profile/jake"
                  >
                                    jake
                  </a>
                  <span
                                    className="date"
                  >
                                    Wed Oct 11 2017
                  </span>
</div>,
                <button
                  className="btn btn-outline-primary btn-sm pull-xs-right"
>
                  <i
                                    className="ion-heart"
                  />
                   
                  27
</button>,
              ],
              "className": "article-meta",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <a
                  href="/profile/jake"
>
                  <img
                                    src="https://i.stack.imgur.com/xHWG8.jpg"
                  />
</a>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    src="https://i.stack.imgur.com/xHWG8.jpg"
/>,
                  "href": "/profile/jake",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <img
                      src="https://i.stack.imgur.com/xHWG8.jpg"
/>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "src": "https://i.stack.imgur.com/xHWG8.jpg",
                    },
                    "ref": null,
                    "rendered": Array [
                      undefined,
                    ],
                    "type": "img",
                  },
                ],
                "type": "a",
              },
              Object {
                "instance": <div
                  class="info"
>
                  <a
                                    href="/profile/jake"
                                    class="author"
                  >
                                    jake
                  </a>
                  <span
                                    class="date"
                  >
                                    Wed Oct 11 2017
                  </span>
</div>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <a
                      className="author"
                      href="/profile/jake"
>
                      jake
</a>,
                    <span
                      className="date"
>
                      Wed Oct 11 2017
</span>,
                  ],
                  "className": "info",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <a
                      href="/profile/jake"
                      class="author"
>
                      jake
</a>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "jake",
                      "className": "author",
                      "href": "/profile/jake",
                    },
                    "ref": null,
                    "rendered": Array [
                      "jake",
                    ],
                    "type": "a",
                  },
                  Object {
                    "instance": <span
                      class="date"
>
                      Wed Oct 11 2017
</span>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Wed Oct 11 2017",
                      "className": "date",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Wed Oct 11 2017",
                    ],
                    "type": "span",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": <button
                  class="btn btn-outline-primary btn-sm pull-xs-right"
>
                  <i
                                    class="ion-heart"
                  />
                  27
</button>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      className="ion-heart"
/>,
                    " ",
                    27,
                  ],
                  "className": "btn btn-outline-primary btn-sm pull-xs-right",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <i
                      class="ion-heart"
/>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "ion-heart",
                    },
                    "ref": null,
                    "rendered": Array [
                      undefined,
                    ],
                    "type": "i",
                  },
                  " ",
                  "27",
                ],
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": <a
              href="/article/test-slug"
              class="preview-link"
>
              <h1>
                            How to build webapps that scale
              </h1>
              <p>
                            This is the description for the post.
              </p>
              <span>
                            Read more...
              </span>
</a>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  How to build webapps that scale
</h1>,
                <p>
                  This is the description for the post.
</p>,
                <span>
                  Read more...
</span>,
              ],
              "className": "preview-link",
              "href": "/article/test-slug",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <h1>
                  How to build webapps that scale
</h1>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "How to build webapps that scale",
                },
                "ref": null,
                "rendered": Array [
                  "How to build webapps that scale",
                ],
                "type": "h1",
              },
              Object {
                "instance": <p>
                  This is the description for the post.
</p>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "This is the description for the post.",
                },
                "ref": null,
                "rendered": Array [
                  "This is the description for the post.",
                ],
                "type": "p",
              },
              Object {
                "instance": <span>
                  Read more...
</span>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Read more...",
                },
                "ref": null,
                "rendered": Array [
                  "Read more...",
                ],
                "type": "span",
              },
            ],
            "type": "a",
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "previews": Array [
          Object {
            "author": Object {
              "image": "https://i.stack.imgur.com/xHWG8.jpg",
              "username": "jake",
            },
            "createdAt": 2017-10-12T00:00:00.000Z,
            "description": "This is the description for the post.",
            "favoritesCount": 27,
            "slug": "test-slug",
            "title": "How to build webapps that scale",
          },
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": Object {
            "instance": <div>
              Feed Picker
</div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Feed Picker",
            },
            "ref": null,
            "rendered": Array [
              "Feed Picker",
            ],
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": <div
            class="article-preview"
>
            <div
                        class="article-meta"
            >
                        <a
                                    href="/profile/jake"
                        >
                                    <img
                                                src="https://i.stack.imgur.com/xHWG8.jpg"
                                    />
                        </a>
                        <div
                                    class="info"
                        >
                                    <a
                                                href="/profile/jake"
                                                class="author"
                                    >
                                                jake
                                    </a>
                                    <span
                                                class="date"
                                    >
                                                Wed Oct 11 2017
                                    </span>
                        </div>
                        <button
                                    class="btn btn-outline-primary btn-sm pull-xs-right"
                        >
                                    <i
                                                class="ion-heart"
                                    />
                                    27
                        </button>
            </div>
            <a
                        href="/article/test-slug"
                        class="preview-link"
            >
                        <h1>
                                    How to build webapps that scale
                        </h1>
                        <p>
                                    This is the description for the post.
                        </p>
                        <span>
                                    Read more...
                        </span>
            </a>
</div>,
          "key": "test-slug",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="article-meta"
>
                <a
                                href="/profile/jake"
                >
                                <img
                                                src="https://i.stack.imgur.com/xHWG8.jpg"
                                />
                </a>
                <div
                                className="info"
                >
                                <a
                                                className="author"
                                                href="/profile/jake"
                                >
                                                jake
                                </a>
                                <span
                                                className="date"
                                >
                                                Wed Oct 11 2017
                                </span>
                </div>
                <button
                                className="btn btn-outline-primary btn-sm pull-xs-right"
                >
                                <i
                                                className="ion-heart"
                                />
                                 
                                27
                </button>
</div>,
              <a
                className="preview-link"
                href="/article/test-slug"
>
                <h1>
                                How to build webapps that scale
                </h1>
                <p>
                                This is the description for the post.
                </p>
                <span>
                                Read more...
                </span>
</a>,
            ],
            "className": "article-preview",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <div
                class="article-meta"
>
                <a
                                href="/profile/jake"
                >
                                <img
                                                src="https://i.stack.imgur.com/xHWG8.jpg"
                                />
                </a>
                <div
                                class="info"
                >
                                <a
                                                href="/profile/jake"
                                                class="author"
                                >
                                                jake
                                </a>
                                <span
                                                class="date"
                                >
                                                Wed Oct 11 2017
                                </span>
                </div>
                <button
                                class="btn btn-outline-primary btn-sm pull-xs-right"
                >
                                <i
                                                class="ion-heart"
                                />
                                27
                </button>
</div>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a
                    href="/profile/jake"
>
                    <img
                                        src="https://i.stack.imgur.com/xHWG8.jpg"
                    />
</a>,
                  <div
                    className="info"
>
                    <a
                                        className="author"
                                        href="/profile/jake"
                    >
                                        jake
                    </a>
                    <span
                                        className="date"
                    >
                                        Wed Oct 11 2017
                    </span>
</div>,
                  <button
                    className="btn btn-outline-primary btn-sm pull-xs-right"
>
                    <i
                                        className="ion-heart"
                    />
                     
                    27
</button>,
                ],
                "className": "article-meta",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <a
                    href="/profile/jake"
>
                    <img
                                        src="https://i.stack.imgur.com/xHWG8.jpg"
                    />
</a>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      src="https://i.stack.imgur.com/xHWG8.jpg"
/>,
                    "href": "/profile/jake",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": <img
                        src="https://i.stack.imgur.com/xHWG8.jpg"
/>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "src": "https://i.stack.imgur.com/xHWG8.jpg",
                      },
                      "ref": null,
                      "rendered": Array [
                        undefined,
                      ],
                      "type": "img",
                    },
                  ],
                  "type": "a",
                },
                Object {
                  "instance": <div
                    class="info"
>
                    <a
                                        href="/profile/jake"
                                        class="author"
                    >
                                        jake
                    </a>
                    <span
                                        class="date"
                    >
                                        Wed Oct 11 2017
                    </span>
</div>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <a
                        className="author"
                        href="/profile/jake"
>
                        jake
</a>,
                      <span
                        className="date"
>
                        Wed Oct 11 2017
</span>,
                    ],
                    "className": "info",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": <a
                        href="/profile/jake"
                        class="author"
>
                        jake
</a>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "jake",
                        "className": "author",
                        "href": "/profile/jake",
                      },
                      "ref": null,
                      "rendered": Array [
                        "jake",
                      ],
                      "type": "a",
                    },
                    Object {
                      "instance": <span
                        class="date"
>
                        Wed Oct 11 2017
</span>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Wed Oct 11 2017",
                        "className": "date",
                      },
                      "ref": null,
                      "rendered": Array [
                        "Wed Oct 11 2017",
                      ],
                      "type": "span",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": <button
                    class="btn btn-outline-primary btn-sm pull-xs-right"
>
                    <i
                                        class="ion-heart"
                    />
                    27
</button>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <i
                        className="ion-heart"
/>,
                      " ",
                      27,
                    ],
                    "className": "btn btn-outline-primary btn-sm pull-xs-right",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": <i
                        class="ion-heart"
/>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "ion-heart",
                      },
                      "ref": null,
                      "rendered": Array [
                        undefined,
                      ],
                      "type": "i",
                    },
                    " ",
                    "27",
                  ],
                  "type": "button",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": <a
                href="/article/test-slug"
                class="preview-link"
>
                <h1>
                                How to build webapps that scale
                </h1>
                <p>
                                This is the description for the post.
                </p>
                <span>
                                Read more...
                </span>
</a>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    How to build webapps that scale
</h1>,
                  <p>
                    This is the description for the post.
</p>,
                  <span>
                    Read more...
</span>,
                ],
                "className": "preview-link",
                "href": "/article/test-slug",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <h1>
                    How to build webapps that scale
</h1>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "How to build webapps that scale",
                  },
                  "ref": null,
                  "rendered": Array [
                    "How to build webapps that scale",
                  ],
                  "type": "h1",
                },
                Object {
                  "instance": <p>
                    This is the description for the post.
</p>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "This is the description for the post.",
                  },
                  "ref": null,
                  "rendered": Array [
                    "This is the description for the post.",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": <span>
                    Read more...
</span>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Read more...",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Read more...",
                  ],
                  "type": "span",
                },
              ],
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
